torch/_inductor/kernel/flex_attention.py

```diff
@@ -93,10 +106,79 @@ def get_float32_precision():
         return "'tf32'"
 
 
+@dataclass(frozen=True)
+class SubgraphOutput:
+    """When building the subgraph buffers we fall into 1 of two cases:
+    1. The subgraph returns a single output node, this happens in the forward and
+        backward pass when there are no input_buffers that require grad
+    2. The subgraph has multiple outputs (i.e. input_buffers that require grad)
+        In this case we want to return these output_nodes from the lowering
```

Comment:
maybe update, since we need both the TensorBox to return as the output from the lowering but the compute buffer to actually inline into the kernel